### PR TITLE
Adding Typedoc for code documentation.

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Resolves [SIMPLY-2162](https://jira.nypl.org/browse/SIMPLY-2162).
Similar to [circulation-web #258](https://github.com/NYPL-Simplified/circulation-web/pull/258), using Typedoc to generate code documentation.

I got pretty far with Storybook and I would prefer it because you can visually show individual components:
Screen Shot 2019-08-19 at 10 44 14 AM
but, there were too many complicated issues when trying to render the stories for the components (many related to context and redux). This is the easiest to do and ties well with the code. For this repo, there is a lot of important non-visual functions that having code documentation for them is useful.

Example: http://nypl-simplified.github.io/opds-web-client/

### CODE REVIEW

**Comment by aslagle** [comment]
There are a bunch of links in the README that don't work in the generated documentation. Most of them are links to classes that have doc pages and should be changed to those - either replaced when the docs are generated or just changed in the README.

**Comment by EdwinGuzman** [comment]
I will change it in the readme because I can link directly using markdown but it will either work in one location and not in the other.

**Comment by aslagle** [comment]
You could use absolute links instead of relative - still weird to have one link to the other place but better than being completely broken.

**Comment by EdwinGuzman** [comment]
Agreed. I updated it and fixed broken react router links.

File: README.md [hunk 1]
```diff
@@ -19,14 +19,14 @@ For an example of OPDS Web Client in use as a standalone app, see the [demo serv
...
-  - `collection`: object representing the current collection data (see `CollectionData` in the [interfaces file](packages/opds-web-client/src/interfaces.ts))
+- `authPlugins`: optional list of objects that implement the [`AuthPlugin` interface](http://nypl-simplified.github.io/opds-web-client/interfaces/authplugin.html). Default: [`BasicAuthPlugin`](http://nypl-simplified.github.io/opds-web-client/globals.html#basicauthplugin)
+- `computeBreadcrumbs`: optional function for customizing breadcrumbs. It defaults to `defaultComputeBreadcrumbs` in the [Breadcrumbs](https://github.com/NYPL-Simplified/opds-web-client/blob/master/packages/opds-web-client/src/interfaces.ts) module, and `hierarchyComputeBreacrumbs` is also available. It should return an array of link objects, each with `url` and `text` properties. Its accepts two arguments:
```
Comment (aslagle) [inline README.md]: This link should still be src/components/Breadcrumbs.tsx

**Pull request merged**